python/pyarrow/tests/parquet/test_data_types.py

```diff
@@ -604,6 +604,25 @@ def test_uuid_extension_type():
         store_schema=False)
 
 
+def test_read_schema_uuid_extension_type(tmp_path):
```

Comment:
I would suggest moving this test to `tests/parquet/test_metadata.py` and also add a check for `read_metadata`.